src/sortStudents.ts

```diff
@@ -1,16 +1,114 @@
 
 export interface Student {
-  // describe Student interface
+  name:string,
+  surname: string,
+  age: number,
+  married:boolean,
+  grades:number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students:Student[],
+  sortBy:SortType,
+  order:SortOrder,
+):Student[] {
+  const studentList:Student[] = students.map(
+    (student:Student) => ({ ...student }),
+  );
```

Comment:
you can do it simpler
```suggestion
  const studentList:Student[] = [...students];
```